src/main/java/com/dutact/web/features/event/student/services/EventServiceImpl.java

```diff
@@ -0,0 +1,36 @@
+package com.dutact.web.features.event.student.services;
+
+import com.dutact.web.core.entities.event.EventStatus;
+import com.dutact.web.core.repositories.EventRepository;
+import com.dutact.web.features.event.student.dtos.EventDto;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Service
+public class EventServiceImpl implements EventService {
+
+    private final EventRepository eventRepository;
+    private final EventMapper eventMapper;
+
+    public EventServiceImpl(EventRepository eventRepository,
+                            EventMapper eventMapper) {
+        this.eventRepository = eventRepository;
+        this.eventMapper = eventMapper;
+    }
+
+    @Override
+    public Optional<EventDto> getEvent(Integer id) {
```

Comment:
Why this method not filter with approved status?

Reply:
Good question